### PR TITLE
MINIFICPP-618: Add C2 triggers, first of which monitors a local file …

### PR DESCRIPTION
**Pull request by phrocker**
…for changes

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/include/c2/C2Trigger.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  C2Trigger(std::string name, utils::Identifier uuid)
+        : core::Connectable(name, uuid){
+
+  }
+  virtual ~C2Trigger() {
+  }
+
+
+  /**
```
[the remark this answers is not in the document]
Comment (phrocker) [reply libminifi/include/c2/C2Trigger.h]: You were right. I went through and added some comments and probably would have missed this had you not mentioned it. Thanks! adding some docs now to our readmes.
Made an early PR to give some people a chance to review and test -- so really appreciate all input!

File: libminifi/include/properties/Properties.h [hunk 2]
```diff
@@ -60,6 +60,9 @@ class Properties {
   // Get the config value
   bool get(std::string key, std::string &value);
 
+  // Get the config value
```
Comment (ottobackwards) [inline libminifi/include/properties/Properties.h]: This should document the alternate keys and why it is different than the above

Comment (phrocker) [reply libminifi/include/properties/Properties.h]: I think you're getting intermediate commits as I work through issues with another Apache person -- this one isn't quite ready for review. I'll close the PR in the meantime.

Comment (ottobackwards) [reply libminifi/include/properties/Properties.h]: sorry

Comment (phrocker) [reply libminifi/include/properties/Properties.h]: No apology, please. It's my fault for opening early!. I truly appreciate the input and hope that when I re-open it you have an opportunity to take a look!

**Pull request closed without merging**